src/couch_replicator_scheduler.erl

```diff
@@ -380,26 +380,58 @@ oldest_job_first(#job{} = A, #job{} = B) ->
     last_started(A) =< last_started(B).
 
 
-not_recently_crashed(#job{} = Job, Now) ->
-    case Job#job.history of
+not_recently_crashed(#job{history = History}, Now) ->
+    case History of
         [{added, _When}] ->
             true;
         [{stopped, _When} | _] ->
             true;
         _ ->
-            case Crashes = crashes_before_stop(Job) of
-                [] ->
-                    true;
-                [{{crashed, _Reason}, When} | _] ->
-                    timer:now_diff(Now, When) >= backoff_micros(length(Crashes))
-            end
+            LatestCrashT = latest_crash_timestamp(History),
+            CrashCount = consecutive_crashes(History),
+            timer:now_diff(Now, LatestCrashT) >= backoff_micros(CrashCount)
     end.
 
--spec crashes_before_stop(#job{}) -> list().
-crashes_before_stop(#job{history = History}) ->
-    EventsBeforeStop = lists:takewhile(
-        fun({Event, _}) -> Event =/= stopped end, History),
-    [Crash || {{crashed, _Reason}, _When} = Crash <- EventsBeforeStop].
+
+% Count consecutive crashes by comparing current and next event pairs. A crash
+% is when a job starts and then crashes in a short period of time. If a job has
+% stopped, consider it healthy. Stop counting crashes after most recent healthy
+% run is encountered ('done' is used as a marker for that).
+-spec consecutive_crashes(history()) -> non_neg_integer().
+consecutive_crashes(History) when is_list(History)->
+    {_, Count} = lists:foldl(fun crashes_fold/2, {nil, 0}, History),
+    Count.
+
+
+-spec crashes_fold(event(), {event() | done, non_neg_integer()}) ->
+    {event() | done, non_neg_integer()}.
+crashes_fold(_Event, {done, Count}) ->
```

Comment:
I started with `takewhile/2` but since we are looking at current and next event, needed an accumulator to keep the additional event.

Perhaps build a new list consisting of pairs of `[{Next2,Current2}, {Next1, Current1}, ...]` and do `takewhile` on that?